### DISCUSSION TITLE
expand_selection when cursor is on a newline marker highlights entire file

### DISCUSSION THREAD

**Discussion by rahil627** [helix-editor/helix#11819] [General]
see title! :)

expand_selection is one of my most use mappings. I think it replicates what double-click does with a mouse. However, many times, i end up highlighting the entire file, pressing escape, jumping back, praying i didn't mess anything up! I expected that it would've expanded the selection on the line.

_note: i wasn't sure if this is really a bug or an enhancement, as it may have been intended, so i filed this under blank issue_

**Comment by the-mikedavis** [comment]
`expand_selection` finds node in the syntax tree that contains a selection and grows the selection to that node's parent. It depends on how the tree-sitter parser for a language treats newlines but they may be considered children of the root of the tree (the whole document) so `expand_selection` would select the whole document

**Reply by rahil627** [reply]
ohhh snap. welp, i guess that's that then.. it depends on each language's tree-sitter parser, and probably don't want to interfere with that..? in my recent case that's precisely the problem: the `.toml` config file, lol. It'll select the entire document unless you are within a `[table]` thing, which i guess it considers a scope/node..

i ran into this problem elsewhere though, in other languages... perhaps there can be an option to prevent it from expanding to the entire document?

it's no biggie, as we can always `shrink` back.. just a bit scary to select all and jump to the end of the document while tappin' away ;)

**Comment by rahil627** [comment]
ooooh snap. i always thought the newline marker was rendered by default for some reason.. :cry: i'm so sorry, can close this.
